### ISSUE TITLE
Changing network input size

### ISSUE THREAD

**Issue by kenrubiooo** [AlexeyAB/darknet#2400]
Hi @AlexeyAB

I am just curious as to how YOLOv3 can easily change its network input size without retraining it. For example, I train the model with 416x416. I can still do inference with 512x512 or 608x608. How? What does it do with the neurons since larger dimension means larger number of neurons?

Thank you!

**Comment by AlexeyAB** [comment]
@kenrubiooo Hi,

Only weights of Fully-Connected layers depend on input/output size.

The weights of Convolutional layers don't depend on input/output size.

Yolo v2/v3 uses Convolutional layers.

**Comment by kenrubiooo** [comment]
Ok. But how does the changing of network input size work wherein I can just instantly change the size without retraining the model on the size that I want? Thank you!

**Comment by AlexeyAB** [comment]
It is solved partially by using `jitter=0.3` that randomly resizes images during training.

And it is solved by using `random=1` that during training randomly resizes network from `1/1.4 + 32` to `1.4 + 32` of initial network resolution. 

So if you trained weights-file by using `width=416 height=416 random=1` then you can use this weights-file for both 320x320 and 608x608 network resolutions. As it is done in default weights-files: https://pjreddie.com/media/files/yolov3.weights

**Comment by kenrubiooo** [comment]
Yes, I read about the random and jitter parameters. I am just curious on how YOLOv3 takes care of the neurons during the change of network input size. Does it remove neurons when it changes from 608x608 to 320x320? What about the weights that are present in the network, i.e. what happens to them during resizing? Thank you!

**Comment by AlexeyAB** [comment]
> What about the weights that are present in the network, i.e. what happens to them during resizing?

Weights are still unchanged. The weights of Convolutional layers don't depend on input/output size.
Convolutional layer uses sliding window - so this window can slide to any width and height of input.

**Comment by kenrubiooo** [comment]
Thank you, @AlexeyAB!

**Comment by kenrubiooo** [comment]
By the way, I tried using the repository of YunYang wherein YOLOv3 is in TensorFlow format. I cannot change the network input size easily, e.g. from 416 to 608. Is this because TF is graph-based, meaning I need to train separately on different network input size? Thank you!
